### ISSUE TITLE
Generator function and custom protocol

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#5235]
**Describe the bug**
Return type of a generator function is incorrectly disallowed to be declared as a custom protocol.

**To Reproduce**
See sample code.

**Expected behavior**
The sample code should type-check.

**Screenshots or Code**
```python
from typing import Protocol

class IntIterator(Protocol):
    def __next__(self, /) -> int: ...

def f() -> IntIterator:
    yield 0
```
![image](https://github.com/microsoft/pyright/assets/15389744/dcd3cdfa-756c-4852-9afc-7768e91bd241)

**VS Code extension or command-line**
Both.

**Additional context**
The actual return type `Generator[Literal[0], Any, None]` itself does conform `IntIterator`.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.312, which I just published. It will also be included in the next insiders release of pylance.
